conda_smithy/schema.py

```diff
@@ -0,0 +1,1310 @@
+
+    Args:
+        enum_class (Union[Type[Enum], Type[EnumMeta]]): A single Enum class or a Union of Enum classes representing the keys for the generated model.
+        value_model (BaseModel): A Pydantic model representing the structure of the values.
+        model_name (str): The name of the generated Pydantic model.
+
+    Example usage:
+
+    ```python
+    class MyEnum1(str, Enum):
+        key1 = "key1"
+        key2 = "key2"
+
+    class MyEnum2(str, Enum):
+        key3 = "key3"
+        key4 = "key4"
+
+    class ValueModel(BaseModel):
+        value_field: int
+
+    # Create an instance of PydanticModelGenerator with a single Enum
+    model_generator1 = PydanticModelGenerator(MyEnum1, ValueModel, "MyGeneratedModel1")
+
+    # Create an instance of PydanticModelGenerator with a Union of Enums
+    model_generator2 = PydanticModelGenerator(Union[MyEnum1, MyEnum2], ValueModel, "MyGeneratedModel2")
+
+    # Retrieve the generated models using the __call__ method
+    GeneratedModel1 = model_generator1()
+    GeneratedModel2 = model_generator2()
+    ```
+    """
+
+    def __init__(
+        self,
+        enum_class: Union[Type[Enum], Type[EnumMeta]],
+        value_model: BaseModel,
+        model_name: str,
+    ):
+        self.enum_class = enum_class
+        self.value_model = value_model
+        self.model_name = model_name
+        self.generated_model = self._generate_model()
+
+    def _generate_model(self) -> BaseModel:
+        field_definitions = {}
+
+        if hasattr(self.enum_class, "__args__"):
+            # Handle Union types
+            for enum_type in self.enum_class.__args__:
+                if isinstance(enum_type, type) and issubclass(enum_type, Enum):
+                    for enum_item in enum_type:
+                        field_definitions[enum_item.value] = (
+                            Optional[self.value_model],
+                            {"description": f"The {enum_item.value} value"},
+                        )
+        else:
+            # Handle single Enum
+            for enum_item in self.enum_class:
+                field_definitions[enum_item.value] = (
+                    Optional[self.value_model],
+                    {"description": f"The {enum_item.value} value"},
+                )
+
+        return create_model(self.model_name, **field_definitions)
+
+    def __call__(self) -> BaseModel:
+        return self.generated_model
+
+
+#############################################
+######## Choices (Enum) definitions #########
+#############################################
+
+
+class CondaBuildTools(str, Enum):
+    conda_build = "conda-build"
+    conda_build_classic = "conda-build+classic"
+    conda_build_mamba = "conda-build+conda-libmamba-solver"
+    mambabuild = "mambabuild"
+
+
+class CIservices(str, Enum):
+    azure = "azure"
+    circle = "circle"
+    travis = "travis"
+    appveyor = "appveyor"
+    default = "default"
```

Comment:
It's a pity that we can't use `StrEnum` til 3.11. Otherwise we could use `auto()` for all the values.